docs/fastdds/dds_layer/core/policy/policy.rst

```diff
@@ -1,5 +1,2259 @@
+
++--------------------------+-------------------------------+-------------------+
+| Data Member Name         | Type                          | Default Value     |
++==========================+===============================+===================+
+| kind                     | :ref:`historyqospolicykind`   | KEEP_LAST         |
++--------------------------+-------------------------------+-------------------+
+| depth                    | int32_t                       | 1                 |
++--------------------------+-------------------------------+-------------------+
+
+* **kind**: Controls if the service should deliver only the most recent values, all the intermediate values or do
+  something in between.
+  See :ref:`historyqospolicykind` for further details.
+* **depth**: Establishes the maximum number of samples that must be kept on the history.
+  It only have effect if the kind is set to KEEP_LAST and it need to be consistent with the
+  :ref:`resourcelimitsqospolicy`, which means that its value must be lower or equal to max_samples_per_instance.
+
+.. note::
+     This QoS Policy concerns to Topic, DataWriter and DataReader entities.
+     :raw-html:`<br />`
+     It cannot be changed after the creation of the entities.
+
+.. _historyqospolicykind:
+
+HistoryQosPolicyKind
+""""""""""""""""""""
+There are two possible values:
+
+* **Keep Last**: The service will only attempt to keep the most recent values of the instance and discard the older
+  ones.
+  The maximum number of samples to keep and deliver is defined by the `depth` of the HistoryQosPolicy, which needs to
+  be consistent with the :ref:`resourcelimitsqospolicy` settings.
+* **Keep All**: The service will attempt to keep all the values of the instance until it can be delivered to all the
+  existing Subscribers.
+  If this option is selected, the depth will not have any effect, so the history is only limited by the values set in
+  :ref:`resourcelimitsqospolicy`.
+  If the limit is reached, the behavior of the system depends on the :ref:`reliabilityqospolicy`, if its kind is
+  BEST_EFFORT the older values will be discarded, but if it is RELIABLE the service blocks the DataWriter until the old
+  values are delivered to all existing Subscribers.
+
+
+Example
+"""""""
+
+C++
+***
+.. literalinclude:: ../../../../../code/DDSCodeTester.cpp
+   :language: c++
+   :dedent: 8
+   :start-after: //DDS_CHANGE_HISTORY_QOS_POLICY
+   :end-before: //!
+
+XML
+***
+.. literalinclude:: /../code/XMLTester.xml
+    :language: xml
+    :start-after: <!-->XML-HISTORY<-->
+    :end-before: <!--><-->
+
+.. _latencybudgetqospolicy:
+
+LatencyBudgetQosPolicy
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. warning::
+    This QoS Policy will be implemented in future releases.
+
+This QoS Policy urgency of the data communication, specifying the maximum acceptable delay from the time the data is
+written until the data is inserted on the cache of the receiver application and notified of the fact.
```

Comment:
Is it until the data is on the cache, or until the application is notified? I would clarify this.
In any case, I would remove the word `cache`, and say something like 'the subscriber receives the data'